addon/ShareProvider.js

```diff
@@ -25,104 +25,20 @@ try {
   SocialService = Cu.import("resource:///modules/SocialService.jsm", {}).SocialService;
 }
 
```

Comment:
The changes in this file are just moving things around so I could reuse the same share code being used by the share toolbar. The diff makes it look like a lot changed, but the actual sharing code didn't change.